etc/ansible/roles/network-runner/providers/dellemc.os9.os9/conf_access_port.yaml

```diff
@@ -1,9 +1,29 @@
 ---
+- name: "dellemc.os9: get running config for interface"
+  dellemc.os9.os9_command:
+    commands: "show running-config interface {{ port_name }}"
+  register: port_description_cmd
+  connection: network_cli
+
+- name: "dellemc.os9: get description from running config"
+  ansible.builtin.set_fact:
+    port_description: "{{ port_description_cmd.stdout[0] | regex_search('(?<=description ).+?(?=[\\n\\r])') }}"
+
 - name: "dellemc.os9: reset interface to default"
   dellemc.os9.os9_config:
     lines:
       - "default interface {{ port_name }}"
   connection: network_cli
+  retries: 5
+  delay: 2
+
+- name: "dellemc.os9: reassign description"
+  dellemc.os9.os9_config:
+    lines:
+      - "description {{ port_description }}"
+    parents: ["interface {{ port_name }}"]
```

Comment:
What does `parents` do here instead of just putting the `interfact {{port_name}}` under lines?

Reply:
Nothing, you can do it either way and it works the same because in OS9 you can switch to an interface config from within another interface config, but I think parents is the documented way to do nested configs like this.